types/header.go

```diff
@@ -30,6 +32,103 @@ type Header struct {
 	Hash         Hash    `json:"hash"`
 }
 
+// HeaderJson represents a block header used for json calls
+type HeaderJSON struct {
```

Comment:
Why does this need to be a separate struct for JSON?

Reply:
Why not define a simple JSON marshal method for the existing struct?

Reply:
I see that you use it in the marshal method for the old `Header` type.

Why not define it locally, why is it exported?

Reply:
because I am using it for both UnmarshalJSON and MarshalJSON I didn't want to have it locally in both methods. What do you think of making it non-exportable

Reply:
It should not be exported outside of the scope you're using it in

Reply:
agreed 4816d822d636264251cac988f4c23f57c2841c34